### PR TITLE
postgres flexible server bicep template

### PR DESCRIPTION
**Pull request by geoberle**
### What this PR does

this PR introduces several bicep modules to
* create a basic postgres flexible server instance and database - `postgres.bicep`
* create a database user based on a managed identity and grant access to the database - `postgres-access.bicep`
* orchestrate the creation of the DB, the user and the placement of configmap to an AKS cluster for CS usage

Jira: https://issues.redhat.com/browse/ARO-7985

### Special notes for your reviewer

the user creation on the database is not supported via ARM, therefore a deploymentscript is currently used to implement this. after discussing the usage of deploymentscripts with MSFT, we will investigate how this can be refactored so that the access management to the database will be done with shell scripts after the bicep run in a way that is compatible with
* production rollouts via EV2
* development environments provisioning via Makefile

cc @whober0521 @nwnt

### CODE REVIEW

File: dev-infrastructure/docs/development-setup.md [hunk 1]
```diff
@@ -346,4 +346,28 @@ To tear down your CS setup:
...
+psql
+```
+
+> The password is a temporary access token that is valid only for 1h
+
+### Access the database from inside of the AKS cluster
+
+* the `cs` namespace of the svc cluster contains
+  * a `ConfigMap` named `cs` with the connection parameters
+  * a `ServiceAccount` named `cs` that is annotated for workload identity usage
```
Comment (markturansky) [inline dev-infrastructure/docs/development-setup.md:370]: can we name these `cs` resources something else?  `cs` is terribly generic

Comment (markturansky) [reply dev-infrastructure/docs/development-setup.md:370]: whatever the current maps/secrets are named in commercial is probably easiest to keep in sync across multiple templates

File: dev-infrastructure/modules/cs.bicep [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    name: postgresServerName
+    databaseAdministrators: [
+      // add the dedicated admin managed identity as administrator
+      // this one is going to be used to manage DB access
+      {
+        principalId: postgresAdminManagedIdentity.properties.principalId
+        principalName: postgresAdminManagedIdentity.name
+        principalType: 'ServicePrincipal'
+      }
+      // use the current user as DB admin if defined - for dev purposes
```
Comment (weinong) [inline dev-infrastructure/modules/cs.bicep:44]: can we add a // TODO: clean up when we release to prod?

Comment (geoberle) [reply dev-infrastructure/modules/cs.bicep]: i removed it entirely from the bicep files.
admin access to the database for dev purposes can now be aquired on-demand via the makefile target `cs-out-of-cluster-db-access`

File: dev-infrastructure/docs/development-setup.md (hunk 1 above)
Comment (geoberle) [reply dev-infrastructure/docs/development-setup.md]: adapted

File: dev-infrastructure/configurations/svc-cluster.bicepparam [hunk 3]
```diff
@@ -11,6 +11,8 @@ param disableLocalAuth = false
 param deployFrontendCosmos = false
 param deployMaestroInfra = false
 param maestroNamespace = 'maestro'
+param deployCsInfra = true
```
Comment (mjlshen) [inline dev-infrastructure/configurations/svc-cluster.bicepparam:14]:
```suggestion
param deployCsInfra = false
```
I think we want this to default to false by convention

**Review by mjlshen — COMMENTED** [review]
Looks good overall, small nit

File: dev-infrastructure/configurations/svc-cluster.bicepparam (hunk 3 above)
Comment (geoberle) [reply dev-infrastructure/configurations/svc-cluster.bicepparam]: good point. fixed

**Review by mjlshen — DISMISSED** [review]

**Review by tonytheleg — APPROVED** [review]

**Review by tonytheleg — APPROVED** [review]

**Pull request merged**